packages/aws-amplify-react-native/src/InAppMessaging/components/FullScreenMessage/FullScreenMessage.tsx

```diff
@@ -11,102 +11,68 @@
-	// values to container View
-	const { containerStyle, wrapperStyle } = useMemo(() => {
-		const { container: baseOverrideContainerStyle } = style ?? {};
-		const { backgroundColor, ...overrideContainerStyle } = Array.isArray(baseOverrideContainerStyle)
-			? StyleSheet.flatten(baseOverrideContainerStyle)
-			: (baseOverrideContainerStyle as ViewStyle) ?? {};
+import { getStyles } from './styles';
+import { FullScreenMessageProps } from './types';
 
-		return {
-			containerStyle: [styles.container, overrideContainerStyle],
+export default function FullScreenMessage(props: FullScreenMessageProps) {
+	const { body, header, image, onClose, primaryButton, secondaryButton } = props;
+	const { renderButtons, renderImage, renderMessage, renderSecondaryButton, styles } = useMessageProps(
+		props,
+		getStyles
+	);
 
-			// prevent passing object with backgroundColor key and undefined or null value
-			// to avoid overriding payload backgroundColor
-			wrapperStyle: [container?.style, backgroundColor ? { backgroundColor } : null],
-		};
-	}, [container?.style, style]);
+	if (!renderMessage) {
+		return null;
+	}
 
-	return shouldDelayMessageRendering ? null : (
-		<MessageWrapper style={wrapperStyle}>
-			<View style={containerStyle}>
+	return (
+		<MessageWrapper style={styles.wrapper}>
+			<View style={styles.container}>
 				<View style={styles.contentContainer}>
 					<IconButton
-						color={style?.closeIconColor}
+						color={styles.iconButton.iconColor}
 						hitSlop={ICON_BUTTON_HIT_SLOP}
 						onPress={onClose}
 						size={ICON_BUTTON_SIZE}
 						source={icons.close}
-						style={[styles.iconButton, style?.closeIconButton]}
+						style={styles.iconButton.container}
 					/>
-					{shouldRenderImage && (
+					{renderImage && (
 						<View style={styles.imageContainer}>
-							<Image source={{ uri: image?.src }} style={imageStyle} />
+							<Image source={{ uri: image?.src }} style={styles.image} />
 						</View>
 					)}
 					<View style={styles.textContainer}>
-						{header?.content && <Text style={[styles.header, header.style, style?.header]}>{header.content}</Text>}
-						{body?.content && <Text style={[styles.message, body.style, style?.message]}>{body.content}</Text>}
+						{header?.content && <Text style={styles.header}>{header.content}</Text>}
+						{body?.content && <Text style={styles.body}>{body.content}</Text>}
 					</View>
 				</View>
-				{(hasPrimaryButton || hasSecondaryButton) && (
+				{renderButtons && (
 					<View style={styles.buttonsContainer}>
-						{hasSecondaryButton && (
+						{renderSecondaryButton && (
 							<Button
 								onPress={secondaryButton.onPress}
-								style={secondaryButtonStyle.container}
-								textStyle={secondaryButtonStyle.text}
+								style={styles.secondaryButton.container}
+								textStyle={styles.secondaryButton.text}
 							>
 								{secondaryButton.title}
 							</Button>
 						)}
-						{hasPrimaryButton && (
+						{renderSecondaryButton && (
```

Comment:
Think this might be a copy/paste bug
```suggestion
						{renderPrimaryButton && (
```